core/src/main/java/ai/timefold/solver/core/impl/localsearch/decider/acceptor/restart/AbstractGeometricRestartStrategy.java

```diff
@@ -0,0 +1,89 @@
+package ai.timefold.solver.core.impl.localsearch.decider.acceptor.restart;
+
+import java.time.Clock;
+
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchMoveScope;
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchPhaseScope;
+import ai.timefold.solver.core.impl.solver.scope.SolverScope;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Restart strategy, which exponentially increases the metric that triggers the restart process.
+ * The first restart occurs after the {@code grace period + 1 * scalingFactor} metric.
```

Comment:
I wonder about the use of the term "grace period" here. We already use it in the new termination.

On one hand, it is the same thing, so it should use the same name.
On the other hand, it is used in a different context, so maybe it shouldn't?
I'm abvialent.

Reply:
The goal is to avoid restarting too early, and I used the term to maintain consistence with the new termination.